### PR TITLE
✨ classic: adds relation initStore

### PR DESCRIPTION
**Pull request by pietrovismara**
- Drops references to `Schema` as the store now can be anything
- Sets the `initStore` callback as `$schema`, so that `Pair` can use it
- Adds a test in `Relation.test.ts`

Unrelated: improves the prefab check in `getRelationTargets`

### CODE REVIEW

File: packages/classic/src/relation/Relation.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import { Schema, World, entityExists, getEntityComponents, registerPrefab, removeEntity } from '..';
-import { ComponentType, defineComponent } from '..';
+import { World, getEntityComponents, registerPrefab } from '..';
 import { $schema } from '../component/symbols';
```
Comment (NateTheGreatt) [inline packages/classic/src/relation/Relation.ts:2]:
```suggestion
```

File: packages/classic/src/relation/types.ts [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
-};
-
 export type RelationTarget = number | string | PrefabToken;
 
-export type RelationType<T extends Schema> = T & {
-	[$pairsMap]: Map<number | string, ComponentType<T>>;
-	[$schema]: Schema;
+export type RelationType<T> = T & {
+	[$pairsMap]: Map<number | string, T>;
+	[$schema]: () => T;
```
Comment (NateTheGreatt) [inline packages/classic/src/relation/types.ts:12]:
```suggestion
	[$initStore]: () => T;
```

File: packages/classic/src/relation/types.ts [hunk 3]
```diff
@@ -1,24 +1,16 @@
 import { $schema } from '../component/symbols';
```
Comment (NateTheGreatt) [inline packages/classic/src/relation/types.ts:1]:
```suggestion
import { $initStore } from '../component/symbols';
```

File: packages/classic/src/relation/Relation.ts [hunk 4]
```diff
@@ -12,52 +11,53 @@ import {
...
-): ComponentType<T> => {
+export const Pair = <T>(relation: RelationType<T>, target: RelationTarget): T => {
 	if (relation === undefined) throw Error('Relation is undefined');
 	if (target === undefined) throw Error('Relation target is undefined');
 	if (target === '*') target = Wildcard;
 
-	const pairsMap = (relation as RelationType<T>)[$pairsMap];
-	const schema = (relation as RelationType<T>)[$schema] as T;
+	const pairsMap = relation[$pairsMap];
+	const schema = relation[$schema];
```
Comment (NateTheGreatt) [inline packages/classic/src/relation/Relation.ts:60]:
```suggestion
	const initStore = relation[$initStore];
```

File: packages/classic/src/relation/Relation.ts [hunk 5]
```diff
@@ -12,52 +11,53 @@ import {
...
 	const pairsMap = new Map();
 	const relation = function (target: RelationTarget) {
 		if (target === undefined) throw Error('Relation target is undefined');
 		if (target === '*') target = Wildcard;
-		return createOrGetRelationComponent<T>(relation, schema, pairsMap, target);
+		return createOrGetRelationComponent<T>(relation, initStore, pairsMap, target);
 	};
 	defineHiddenProperty(relation, $pairsMap, pairsMap);
-	defineHiddenProperty(relation, $schema, schema);
+	defineHiddenProperty(relation, $schema, initStore);
```
Comment (NateTheGreatt) [inline packages/classic/src/relation/Relation.ts:47]:
```suggestion
	defineHiddenProperty(relation, $initStore, initStore);
```

**Review by NateTheGreatt — CHANGES_REQUESTED** [review]
great work thanks for this! couple change requests

**Comment by krispya** [comment]
Looks good to me. I'm merging.

**Pull request merged**